android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/LanguageResource.java

```diff
@@ -45,5 +64,71 @@ public int hashCode() {
 
   public abstract Bundle buildDownloadBundle();
 
-  //public abstract boolean equals();
+  /**
+   * Constructor using JSON Objects from installed language resource list
+   * @param installedObj
+   */
+  public LanguageResource(JSONObject installedObj) {
+    this.fromJSON(installedObj);
+  }
+
+  public LanguageResource(JSONObject languageJSON, JSONObject keyboardJSON) {
+    // Noop - needed for Keyboard()
+  }
+
+  public LanguageResource(JSONObject lexicalModelJSON, boolean fromCloud) {
+    // Noop - needed for LexicalModel()
+  }
```

Comment:
These aren't needed because this class is abstract and thus never directly instantiated.